api/pkg/apis/v1alpha1/managers/solution/solution-manager.go

```diff
@@ -577,11 +584,24 @@ func (s *SolutionManager) getTargetStateForStep(step model.DeploymentStep, deplo
 	return targetSpec
 }
 
-func (s *SolutionManager) saveSummary(ctx context.Context, objectName string, generation string, hash string, summary model.SummarySpec, state model.SummaryState, namespace string) {
+func (s *SolutionManager) saveSummary(ctx context.Context, objectName string, generation model.DeploymentGeneration, hash string, summary model.SummarySpec, state model.SummaryState, namespace string) error {
 	// TODO: delete this state when time expires. This should probably be invoked by the vendor (via GetSummary method, for instance)
-	log.DebugfCtx(ctx, " M (Solution): saving summary, objectName: %s, state: %s, namespace: %s, targetCount %d, successCount %d",
-		objectName, state, namespace, summary.TargetCount, summary.SuccessCount)
-	s.StateProvider.Upsert(ctx, states.UpsertRequest{
+	log.DebugfCtx(ctx, " M (Solution): saving summary, objectName: %s, state: %v, namespace: %s, generation %v, hash %s, targetCount %d, successCount %d",
+		objectName, state, namespace, generation, hash, summary.TargetCount, summary.SuccessCount)
+
+	oldSummary, err := s.GetSummary(ctx, objectName, namespace)
+	if err != nil && !v1alpha2.IsNotFound(err) {
+		log.ErrorfCtx(ctx, " M (Solution): failed to get previous summary: %+v", err)
```

Comment:
For the first iteration the older summary won't be found, right? Should we skip check in this case instead of returning error?

Reply:
Yes, you are right. I use not IsNotFound in this condition to prevent return error for NotFound error.